agent/agent_test.go

```diff
@@ -1026,8 +1025,6 @@ func TestAgent_IndexChurn(t *testing.T) {
 // verifyIndexChurn registers some things and runs anti-entropy a bunch of times
 // in a row to make sure there are no index bumps.
 func verifyIndexChurn(t *testing.T, tags []string) {
-	t.Helper()
```

Comment:
Why did you remove this? Generally anything that isn't a `func TestABC(t *testing.T)` function along the way that can fail the test should have the `Helper()` called to make it easier to follow failure line and file numbers.

Reply:
> Generally anything that isn't a func TestABC(t *testing.T) function along the way that can fail the test should have the Helper() called to make it easier to follow failure line and file numbers.

Right, generally that is true, because in general most functions called with a `testing.T` are test helpers. In this case the function is not a test helper. 

https://github.com/hashicorp/consul/blob/4563fe4064acfebb1d84ad54561b989b98e6e34a/agent/agent_test.go#L1014-L1023

This function is called in two places from `TestAgent_IndexChurn`. As we can see, `TestAgent_indexChurn` does not actually contain any logic, it is effectively a table test where the test cases are implemented by a parameterized function `verifyIndexChurn`.

If we put a `t.Helper()` in `verifyIndexChurn` then the reported test line is useless. It doesn't tell us anything more than the test name because each subtest is online a single line.

By removing the `t.Helper()` we get the relevant line in the test case function.

A more appropriate name for `verifyIndexChurn` might be `testIndexChurnWithTags()`, which would probably make this distinction (of test helper vs test case) more obvious.

Reply:
OK.